pkg/operatorstatus/monitor.go

```diff
@@ -0,0 +1,215 @@
+package operatorstatus
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+const (
+	// clusterOperatorName is the name of the ClusterOperator.
+	clusterOperatorName = "marketplace"
+
+	// minimumSyncs represents the minimum number of syncs Marketplace must see
+	// prior to reporting that it is available to the ClusterOperator.
+	minimumSyncs uint = 2
+
+	// degradedRatio is the ratio of failed transition to syncs that must be reached
+	// prior to reporting that the marketplace operator is  degraded.
+	degradedRatio = 0.7
+)
+
+var (
+	instance *statusMonitor
```

Comment:
Since this is a singleton, can we have a better name? `instance` doesn't make sense out of context. What about `statusMonitorSingleton` or `statusMon`?

Reply:
I like `statusMon`, we can put a comment about it being a singleton above it's declaration.